### ISSUE TITLE
`Mode::from_raw_bits` does not mask out `S_IFMT`

### ISSUE THREAD

**Issue by rusty-snake** [bytecodealliance/rustix#1104]
[Rustix documentation for `statat`](https://docs.rs/rustix/0.38.34/rustix/fs/fn.statat.html):

> fstatat(dirfd, path, flags)—Queries metadata for a file or directory.
>
> [Mode::from_raw_mode](https://docs.rs/rustix/0.38.34/rustix/fs/struct.Mode.html#method.from_raw_mode) and [FileType::from_raw_mode](https://docs.rs/rustix/0.38.34/rustix/fs/enum.FileType.html#method.from_raw_mode) may be used to interpret the st_mode field.

What I am basically doing is

```rust
statat(dirfd, path, AtFlags::SYMLINK_NOFOLLOW).is_ok_and(|s| {
    FileType::from_raw_mode(s.st_mode) == FileType::Directory // OT: Here I would wish a .is_dir() helper
    && Mode::from_raw_mode(s.st_mode) == Mode::RWXU
})
```

where the first check on FileType works as you would expect. The second chekc on Mode however took me some long debugging session to realize that the left hand side is actually `Mode::RWXU.from_bits_retain(FileType::Directory.as_raw_mode())`. My expectation, influenced by the documentation above, was that `S_IFMT` gets masked out of course.

**Comment by sunfishcode** [comment]
I think your expectation is correct. I've now fixed this in #1124.

**Comment by sunfishcode** [comment]
This is fixed in rustix 0.38.35.
